fuel-vm/src/interpreter/memory.rs

```diff
@@ -963,39 +981,27 @@ impl OwnershipRegisters {
             return false
         }
 
-        let heap_end = if self.context.is_external() {
-            VM_MAX_RAM
-        } else {
-            self.prev_hp
-        };
-
-        self.hp != heap_end && range.end <= heap_end
+        self.hp != self.prev_hp && range.end <= self.prev_hp
     }
 }
 
 /// Attempt copy from slice to memory, filling zero bytes when exceeding slice boundaries.
 /// Performs overflow and memory range checks, but no ownership checks.
-pub(crate) fn copy_from_slice_zero_fill_noownerchecks<A: ToAddr, B: ToAddr>(
+pub(crate) fn copy_from_slice_zero_fill<A: ToAddr, B: ToAddr>(
     memory: &mut MemoryInstance,
+    owner: OwnershipRegisters,
     src: &[u8],
     dst_addr: A,
     src_offset: usize,
     len: B,
 ) -> SimpleResult<()> {
-    let range = memory.verify(dst_addr, len)?;
+    let range = memory.write(owner, dst_addr, len)?;
 
     let src_end = src_offset.saturating_add(range.len()).min(src.len());
     let data = src.get(src_offset..src_end).unwrap_or_default();
-    let (r_data, r_zero) = range.split_at_offset(data.len());
-
-    memory
```

Comment:
Hmm, maybe we need to revisit all places where we use `write_noownerchecks`, maybe we don't need it anymore and we can just pass `OwnershipRegisters`